Mathlib/Tactic/Linter/DeprecatedSyntaxLinter.lean

```diff
@@ -89,11 +121,13 @@ replacement syntax. For each individual case, linting can be turned on or off se
 * `refine'`, superseded by `refine` and `apply` (controlled by `linter.style.refine`)
 * `cases'`, superseded by `obtain`, `rcases` and `cases` (controlled by `linter.style.cases`)
 * `admit`, superseded by `sorry` (controlled by `linter.style.admit`)
+* `set_option maxHeartbeats`, should contain a comment (controlled by `linter.style.maxHeartbeats`)
 -/
-def deprecatedSyntaxLinter : Linter where run := withSetOptionIn fun stx => do
+def deprecatedSyntaxLinter : Linter where run stx := do
```

Comment:
The reason for this change is that otherwise leading `set_option`s would be invisible to the linter.

Reply:
Makes sense. Does that effect a change to the other linters; i.e. does a `set_option linter.refine true in` still have the same effect? (Is that tested already?)

Reply:
There are tests for the other linters, but they should not be affected, since they do not check a `set_option ... in` syntax.

Reply:
Yes, but I thought the `withSetOptionIn` had the effect of enabling/disabling the linter if it was inside an (e.g.) `set_option linter.style.refine false in ...` block. Does that still work (say, because the `Linter.getLinterValue` picks up the option change)?